pkg/util/chunk/chunk_in_disk.go

```diff
@@ -0,0 +1,316 @@
+			return
+		}
+	}
+
+	serializedBytesNum := d.serializeDataToBuf(chk)
+
+	var writeNum int
+	writeNum, err = d.dataFile.write(d.buf)
+	if err != nil {
+		return
+	}
+
+	if int64(writeNum) != serializedBytesNum {
+		return errors2.New("Some data fail to be spilled to disk")
+	}
+	d.offsetOfEachChunk = append(d.offsetOfEachChunk, d.totalDataSize)
+	d.totalDataSize += serializedBytesNum
+	d.totalRowNum += int64(chk.NumRows())
+	d.dataFile.offWrite += serializedBytesNum
+
+	d.diskTracker.Consume(serializedBytesNum)
+	return
+}
+
+func (d *DataInDiskByChunks) getChunkSize(chkIdx int) int64 {
+	totalChunkNum := len(d.offsetOfEachChunk)
+	if chkIdx == totalChunkNum-1 {
+		return d.totalDataSize - d.offsetOfEachChunk[chkIdx]
+	}
+	return d.offsetOfEachChunk[chkIdx+1] - d.offsetOfEachChunk[chkIdx]
+}
+
+// GetChunk gets a Chunk from the DataInDiskByChunks by chkIdx.
+func (d *DataInDiskByChunks) GetChunk(chkIdx int) (*Chunk, error) {
+	reader := d.dataFile.getSectionReader(d.offsetOfEachChunk[chkIdx])
+	chkSize := d.getChunkSize(chkIdx)
+
+	if cap(d.buf) < int(chkSize) {
+		d.buf = make([]byte, chkSize)
+	} else {
+		d.buf = d.buf[:chkSize]
+	}
+
+	readByteNum, err := io.ReadFull(reader, d.buf)
+	if err != nil {
+		return nil, err
+	}
+
+	if int64(readByteNum) != chkSize {
+		return nil, errors2.New("Fail to restore the spilled chunk")
+	}
+
+	chk := NewEmptyChunk(d.fieldTypes)
+	d.deserializeDataToChunk(chk)
+
+	return chk, nil
+}
+
+// Close releases the disk resource.
+func (d *DataInDiskByChunks) Close() {
+	if d.dataFile.file != nil {
+		d.diskTracker.Consume(-d.diskTracker.BytesConsumed())
+		terror.Call(d.dataFile.file.Close)
+		terror.Log(os.Remove(d.dataFile.file.Name()))
+	}
+}
+
+func (d *DataInDiskByChunks) serializeColMeta(pos int64, length int64, nullMapSize int64, dataSize int64, offsetSize int64) {
+	// TODO we can replace these with spill util in the future
+	*(*int64)(unsafe.Pointer(&d.buf[pos])) = length
+	*(*int64)(unsafe.Pointer(&d.buf[pos+int64Len])) = nullMapSize
+	*(*int64)(unsafe.Pointer(&d.buf[pos+int64Len*2])) = dataSize
+	*(*int64)(unsafe.Pointer(&d.buf[pos+int64Len*3])) = offsetSize
+}
+
+func (d *DataInDiskByChunks) deserializeColMeta(pos *int64) (length int64, nullMapSize int64, dataSize int64, offsetSize int64) {
+	// TODO we can replace these with spill util in the future
+	length = *(*int64)(unsafe.Pointer(&d.buf[*pos]))
+	*pos += int64Len
+
+	nullMapSize = *(*int64)(unsafe.Pointer(&d.buf[*pos]))
+	*pos += int64Len
+
+	dataSize = *(*int64)(unsafe.Pointer(&d.buf[*pos]))
+	*pos += int64Len
+
+	offsetSize = *(*int64)(unsafe.Pointer(&d.buf[*pos]))
+	*pos += int64Len
+	return
+}
+
+func (d *DataInDiskByChunks) serializeOffset(pos *int64, offsets []int64, offsetSize int64) {
+	d.buf = d.buf[:*pos+offsetSize]
+	for _, offset := range offsets {
+		*(*int64)(unsafe.Pointer(&d.buf[*pos])) = offset
+		*pos += int64Len
+	}
+}
+
+func (d *DataInDiskByChunks) serializeChunkData(pos *int64, chk *Chunk, selSize int64) {
```

Comment:
better?
```suggestion
func (d *DataInDiskByChunks) serializeChunkMeta(pos *int64, chk *Chunk, selSize int64) {
```

Reply:
> better?

I think data is better as data serialized in this function are data fields that belong to the Chunk.